include/BALL/KERNEL/bond.iC

```diff
@@ -32,12 +32,20 @@ void Bond::destroy()
 BALL_INLINE
 void Bond::finalize()
 {
-	if (*first_ > *second_)
-	{
-		Atom* tmp = first_;
-		first_ = second_;
-		second_ = tmp;
-	}
+        if (first_ && second_)
+        {
+                if (*first_ > *second_)
+                {
+                        Atom* tmp = first_;
+                        first_ = second_;
+                        second_ = tmp;
+                }
+        }
+        else
```

Comment:
While I am certain that this fixes the segmentation fault I am not sure whether this is a good thing. It should be a programming error to create a bond that has a NULL atom and if this is detected the program should crash. With your fix the problem crashed later as soon as the respective member of bond is dereferenced.

Also remember that each Atom known which Bonds it is a part of. By setting the respective member of the Bond to `nullptr` this creates an inconsistency which may bite you later.

In my opinion this code *should* crash. Ideally, we would add an assertion at the beginning of the method, which ensures that both atoms are valid.